File: Practice/atcoder/ABC/043/src/c.py
# まさかの
# 問題文をちゃんと読もう 文ってか制約
n = int(input())
arr = list(map(int, input().split()))
ans = 10e9
for t in range(-100, 101):
    ans_tmp = 0
    for v in arr:
        ans_tmp += abs(t - v)**2
    ans = min(ans, ans_tmp)

print(ans)

File: Practice/atcoder/ABC/043/src/c.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)
	ans := 1<<32 - 1
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		var m int
		fmt.Scan(&m)
		arr[i] = m
	}
	for i := -100; i <= 100; i++ {
		tmp := 0
		for _, ai := range arr {
			tmp += (i - ai) * (i - ai)
		}
		ans = min(ans, tmp)
	}
	fmt.Println(ans)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}
